test/unittest/tests/block/test_block_addr.cpp

```diff
@@ -0,0 +1,174 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+// This file unit tests block manager functions relating to packing and unpacking address cookies.
+
+#include "wt_internal.h"
+#include <catch2/catch.hpp>
+#include <iostream>
+#include <vector>
+
+void
+unpack_addr_cookie_and_check(const uint8_t *packed, uint32_t block_allocsize, wt_off_t pack_offset,
+  uint32_t pack_size, uint32_t pack_checksum)
+{
+    uint64_t o = 0, s = 0, c = 0;
+    REQUIRE(__wt_vunpack_uint(&packed, 0, &o) == 0);
+    REQUIRE(__wt_vunpack_uint(&packed, 0, &s) == 0);
+    REQUIRE(__wt_vunpack_uint(&packed, 0, &c) == 0);
+
+    // Adjust the unpacked values as the block manager also does this to avoid storing large
+    // offsets.
+    wt_off_t unpacked_offset = (s == 0) ? (wt_off_t)o : (wt_off_t)(o + 1) * block_allocsize;
+    uint32_t unpacked_size = (s == 0) ? (uint32_t)s : (uint32_t)s * block_allocsize;
+    uint32_t unpacked_checksum = (uint32_t)c;
+
+    if (pack_size != 0) {
+        CHECK(unpacked_offset == pack_offset);
```

Comment:
Do we have any guidance on when to use CHECK instead of REQUIRE? 

cc: @jeremythorp

Reply:
The short version is that you can always use REQUIRE, but sometimes CHECK is better.
The key difference is that REQUIRE will test the expression and abort the test case if the expression was false. On the other hand, CHECK will report the error, but keep going.
So, if you are checking if a pointer is null before dereferencing it, then you have to use REQUIRE. But if you're checking a that a numerical result is as expected, it may make more sense to continue after reporting the error.
These lines of code are great examples where CHECK is actually better.
For more info, see the docs: https://github.com/catchorg/Catch2/blob/devel/docs/assertions.md (which I highly recommend reading).